src/component/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,128 @@
+import { useContext } from 'react';
+import { DispatchContext, StateContext } from '../../store/store';
+import cn from 'classnames';
+import { IsUseTodos } from '../../types/IsUseTodos';
+import { deleteTodo } from '../../api/todos';
+
+export const TodoList = () => {
+  const dispatch = useContext(DispatchContext);
+  const { todos, useTodos, changerId, idTodoSubmitting, vaitTodoId } =
+    useContext(StateContext);
+
+  const todosFilter = todos.filter(todo => {
+    switch (useTodos) {
+      case IsUseTodos.Active:
+        return !todo.completed;
+
+      case IsUseTodos.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const handleDeleteTodo = (id: number) => {
+    dispatch({ type: 'setVaitTodoId', id: id });
+
+    deleteTodo(id)
+      .then(() => {
+        dispatch({ type: 'removeTodo', id: id });
+      })
+      .catch(() => {
+        dispatch({ type: 'setError', error: 'Unable to delete a todo' });
+      })
+      .finally(() => {
+        dispatch({ type: 'deleteVaitTodoId', id: id });
+      });
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todosFilter.map(({ id, title, completed }) => (
+        <div
+          data-cy="Todo"
+          className={cn('todo', { completed: completed })}
+          key={id}
+        >
+          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={completed}
+              onChange={() => dispatch({ type: 'SetCheckedTodo', id: id })}
+            />
+          </label>
+
+          {id !== changerId && (
+            <span
+              data-cy="TodoTitle"
+              className="todo__title"
+              onDoubleClick={() =>
+                dispatch({ type: 'setChangedTodoId', id: id })
+              }
+            >
+              {title}
+            </span>
+          )}
+
+          {id === changerId && (
+            <form
+              onSubmit={e => {
+                e.preventDefault();
+                dispatch({ type: 'setChangedTodoId', id: 0 });
+              }}
```

Comment:
Pls move the function inside `onSubmit` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**